src/services/chris_integration.tsx

```diff
@@ -238,119 +238,164 @@ class ChrisIntegration {
-              severity: content['Geographic severity'],
-              extentScore: content['Geographic extent score']
-            }
-            newSeries.opacity = {
-              severity: content['Opacity severity'],
-              extentScore: content['Opacity extent score']
-            }
-          } else if (!fileObj.data.fname.includes('json')) {
-            newSeries.imageId = fileObj.data.id;
-            
-            // Fetch image URL
-            if (newSeries.imageId) {
-              const imgBlob = await DicomViewerService.fetchImageFile(newSeries.imageId);
-              const urlCreator = window.URL || window.webkitURL;
-              newSeries.imageUrl = urlCreator.createObjectURL(imgBlob);
+              newSeries.opacity = {
+                severity: content['Opacity severity'],
+                extentScore: content['Opacity extent score']
+              }
+            } else if (!fileObj.data.fname.includes('json')) {
+              newSeries.imageId = fileObj.data.id;
+              
+              // Fetch image URL
+              if (newSeries.imageId) {
+                const imgBlob = await DicomViewerService.fetchImageFile(newSeries.imageId);
+                const urlCreator = window.URL || window.webkitURL;
+                newSeries.imageUrl = urlCreator.createObjectURL(imgBlob);
+              }
+  
+              // get dcmImageId from dircopy
+              const dircopyPlugin = pluginlists[pluginlists.findIndex((plugin: any) => plugin.data.plugin_name === PluginModels.Plugins.FS_PLUGIN)]
+              const dircopyFiles = (await dircopyPlugin.getFiles({
+                limit: 100,
+                offset: 0
+              })).data;
+              const dcmImageFile = dircopyFiles[dircopyFiles.findIndex((file: any) => file.fname.includes('.dcm'))]
+              newSeries.imageName = dcmImageFile.fname;
             }
-
-            // get dcmImageId from dircopy
-            const dircopyPlugin = pluginlists[pluginlists.findIndex((plugin: any) => plugin.data.plugin_name === PluginModels.Plugins.FS_PLUGIN)]
-            const dircopyFiles = (await dircopyPlugin.getFiles({
-              limit: 100,
-              offset: 0
-            })).data;
-            const dcmImageFile = dircopyFiles[dircopyFiles.findIndex((file: any) => file.fname.includes('.dcm'))]
-            newSeries.imageName = dcmImageFile.fname;
           }
+          
+          if (studyInstance) studyInstance.series.push(newSeries)
         }
-        if (studyInstance) studyInstance.series.push(newSeries)
       }
+
+      // Update fetchLimit to be remaining number of StudyInstanceWithSeries desired
+      fetchLimit = limit + 1 - pastAnalysis.length
     }
-    return pastAnalysis;
+
+    // If the end of Feeds was reached, return all collected StudyInstanceWithSeries, otherwise, discard the last extra Study
+    const pastAnalysesToReturn = isAtEndOfFeeds ? pastAnalysis : pastAnalysis.slice(0, -1);
```

Comment:
How come it will return an extra study at the end?

Reply:
To confirm: we either return all that is left, or we return exactly the amount that was requested?

Reply:
The way the getPastAnalysis works is it aims to collect one extra row than needed for the current page. Doing this allows us to ensure that if there is a row at the bottom of the page that contains multiple Series grouped together, that the Series won't spill into the next page. The extra row can be discarded before returning.

The exception to this is if we've reached the end of Feeds on Swift (i.e. there is no way to populate the page to its full size because there are no more analyses to do so). If we've reached the end, then we can just return the entire pastAnalysis array, since there definitely won't be an extra. This is what the ternary is for.